syserol/impute.py

```diff
@@ -2,26 +2,30 @@
 
 import numpy as np
 import tensorly as tl
+from statsmodels.multivariate.pca import PCA
 from .dataImport import createCube
 from .tensor import perform_CMTF
 
 
-def evaluate_missing(comps, numSample, chords):
-    """ check differences between original and recon values for different number of components.
-    chords: whether to leave out tensor chords or individual values """
+def flatten_to_mat(tensor, matrix):
+    n = tensor.shape[0]
+    tMat = np.reshape(tensor, (n, -1))
+    tMat = tMat[:, ~np.all(np.isnan(tMat), axis=0)]
```

Comment:
One thing I was worried about with flattening the cube into the matrix was this line. Is this going to do something weird to our imputed values, or no?

Reply:
If all measurement is removed by chance, then that chord will be removed too, if that is the catch you refer to

Reply:
Yes, I was thinking if an entire chord gets removed by chance somehow then we might be comparing the cube and matrix factorizations incorrectly? But it might not be an issue since we are just removing individual values.